src/rules/selector-max-specificity/index.js

```diff
@@ -0,0 +1,46 @@
+// Bring in dependencies
+import { isNumber } from "lodash"
+import { calculate }  from "specificity"
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-max-specificity"
+
+export const messages = ruleMessages(ruleName, {
+  expected: (selector, specificity) => `Expected "${selector}" to have a specificity equal to or less than "${specificity}"`,
+})
+
+export default function (max) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: max,
+      possible: [isNumber],
+    })
+
+    if (!validOptions) { return }
+
+    root.walkRules(checkSpecificity)
+
+    function checkSpecificity(rule) {
+      // using rule.selectors gets us each selector in the eventuality we have a comma separated set
+      rule.selectors.forEach(function (selector) {
+        let computedSpecificity = calculate(selector)[0].specificity.replace(/,/g, "")
```

Comment:
Maybe coerce the specificity to a number before the comparison (`const computedSpecifity = Number(..)`), so that if anything goes haywire we might end up with less confusing debugging?